tasks/network_tracer.py

```diff
@@ -145,41 +180,107 @@ def codegen(ctx):
     ctx.run("easyjson {}".format(path))
 
 
-def build_ebpf_builder(ctx):
-    """build_ebpf_builder builds the docker image for the ebpf builder
-    """
+@task
+def object_files(ctx, install=True):
+    """object_files builds the eBPF object files"""
+    build_object_files(ctx, install=install)
 
-    cmd = "docker build -t {image} -f {file} ."
 
-    if should_use_sudo(ctx):
-        cmd = "sudo " + cmd
+def build_object_files(ctx, install=True):
+    """build_object_files builds only the eBPF object
+    set install to False to disable replacing the assets
+    """
 
-    ctx.run(cmd.format(image=EBPF_BUILDER_IMAGE, file=EBPF_BUILDER_FILE))
+    headers_dir = "/usr/src"
+    linux_headers = [
+        os.path.join(headers_dir, d) for d in os.listdir(headers_dir)
+        if "linux-headers" in d
+    ]
+
+    bpf_dir = os.path.join(".", "pkg", "ebpf")
+    c_dir = os.path.join(bpf_dir, "c")
+
+    flags = [
+        '-D__KERNEL__',
+        '-D__ASM_SYSREG_H',
+        '-D__BPF_TRACING__',
+        '-Wno-unused-value',
+        '-Wno-pointer-sign',
+        '-Wno-compare-distinct-pointer-types',
+        '-Wunused',
+        '-Wall',
+        '-Werror',
+        '-O2',
+        '-emit-llvm',
+        '-c',
+        os.path.join(c_dir, "tracer-ebpf.c"),
+    ]
+
+    subdirs = [
+        "include",
+        "include/uapi",
+        "include/generated/uapi",
+        "arch/x86/include",
```

Comment:
Do we intercept any x86-specific functions in our ebpf program? I'm curious about why we need to include arch-dependent kernel parts.

Reply:
Actually, it might be complicate to install this on our builder images...

Reply:
I just tried building without those, and it looks like inluding `<linux/bpf.h>` needs them. We will figure something out if we want to target ARM at some point :)